tensorboard/webapp/widgets/source_code/source_code_component.ts

```diff
@@ -46,9 +46,8 @@ export class SourceCodeComponent implements OnInit, OnChanges, OnDestroy {
   @Input()
   focusedLineno: number | null = null;
 
-  // TODO(cais): Explore better typing by depending on external libraries.
   @Input()
-  monaco: any | null = null;
+  monaco: typeof monaco | null = null;
```

Comment:
Is this referring to an ambient monaco variable? :\ Hate how it does that :(

Reply:
Yup.  Unfortunately the official `monaco.d.ts` uses TS Namespaces for a few top-level things, like `monaco` and `monaco.editor`.  I couldn't find other projects that pass around the `window.monaco` object around directly _and_ want to properly type it.  The only way I found to type those was to use this typeof keyword, as suggested by a StackOverflow post.

https://stackoverflow.com/questions/42815932/use-namespace-as-type-in-typescript